app/models/endpoint.rb

```diff
@@ -9,6 +9,30 @@ class Endpoint < ApplicationRecord
   validates :url, :uniqueness => true, :if => :url
   validate :validate_certificate_authority
 
+  after_create  :ensure_monitoring_manager_creation
+  after_destroy :ensure_monitoring_manager_deletion
+
+  def ensure_monitoring_manager_creation
+    ems = resource
+    monitoring_manager = ems.try(:monitoring_manager)
+
+    # Make sure monitoring manager is deleted\created for the prometheus endpoint
+    if role == "prometheus_alerts" && monitoring_manager.nil?
```

Comment:
Nice.
To keep Endpoint class mostly provider-agnostic, consider keeping the logic in HasMonitoringManagerMixin method(s), with the hooks here reduced to something like
```
ems.endpoint_created(role) if ems.responds_to?(:endpoint_created)
```